packages/integration/src/compile.ts

```diff
@@ -46,7 +51,12 @@ export async function compile({
     write: false,
     plugins: [vanillaExtractFilescopePlugin()],
     absWorkingDir: cwd,
-  });
+  };
+
+
+  const resolvedOptions = merge(baseOptions, esbuildOptions ?? {});
```

Comment:
Could we only run the merge if there's anything to merge?

e.g. 
```suggestion
  const resolvedOptions = esbuildOptions ? merge(baseOptions, esbuildOptions) : baseOptions;
```

Reply:
Also just thinking about this, I'm wondering if exposing the whole set of options is a good idea. Outside of plugins, what can you add here that won't break vanilla-extract. Maybe we should just be exposing plugins? Thoughts?

Reply:
Scoping is probably sensible. Perhaps these options would make sense:
`plugins`, `external`, `define`, `loader`?